app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/TestUtils.java

```diff
@@ -125,4 +128,100 @@ public static void deleteTestBook(Solo solo, Resources resources, int title)
         /* Wait for the MyBooks activity after deleting the book */
         solo.waitForActivity(MyBooksActivity.class, (int) SHORT_WAIT);
     }
+
+    public static class BookManager
+    {
+        private Book availableBook;
+        private Book acceptedBook;
+        private Book requestedBook;
+        private Book borrowedBook;
+        private Resources resources;
+        private Solo solo;
+
+        public BookManager(Solo solo, Resources resources)
+        {
+            this.solo = solo;
+            this.resources = resources;
+
+            /* Private book objects so we can easily add and remove from firebase */
+            availableBook = new Book(resources.getString(R.string.test_account1_username),
+                    resources.getString(R.string.test_book_available_title),
+                    resources.getString(R.string.test_book_available_author),
+                    resources.getString(R.string.test_book_available_description),
+                    resources.getString(R.string.test_book_isbn),
+                    Status.Available,
+                    "");
+            acceptedBook = new Book(resources.getString(R.string.test_account1_username),
+                    resources.getString(R.string.test_book_accepted_title),
+                    resources.getString(R.string.test_book_accepted_author),
+                    resources.getString(R.string.test_book_accepted_description),
+                    resources.getString(R.string.test_book_isbn),
+                    Status.Accepted,
+                    "");
+            requestedBook = new Book(resources.getString(R.string.test_account1_username),
+                    resources.getString(R.string.test_book_requested_title),
+                    resources.getString(R.string.test_book_requested_author),
+                    resources.getString(R.string.test_book_requested_description),
+                    resources.getString(R.string.test_book_isbn),
+                    Status.Requested,
+                    "");
+            /* Set the second test account as the requested */
+            requestedBook.addRequester(resources.getString(R.string.test_account2_username));
+            borrowedBook = new Book(resources.getString(R.string.test_account1_username),
+                    resources.getString(R.string.test_book_borrowed_title),
+                    resources.getString(R.string.test_book_borrowed_author),
+                    resources.getString(R.string.test_book_borrowed_description),
+                    resources.getString(R.string.test_book_isbn),
+                    Status.Borrowed,
+                    "");
```

Comment:
Current Implementation keeps the borrower in the requested array and removes others for a borrowed book. 
We should add the borrower to the requester array here, right?

Reply:
Done

Reply:
sorry, but this should be the same for all books that are not available, eg accepted and the intermediate status ones if you add in future

Reply:
what should be the same? I'm not sure I understand what you mean

Reply:
The accepted book should have the borrower added to the requesters list